firefox-ios/Storage/Rust/RustLogins.swift

```diff
@@ -829,6 +830,30 @@ public class RustLogins: LoginsProtocol {
         return deferred
     }
 
+    func updateLogin(id: String, login: LoginEntry, completionHandler: @escaping () -> Void) {
+        queue.async {
+            guard self.isOpen else {
+                _ = LoginsStoreError.UnexpectedLoginsApiError(reason: "Database is closed")
+                completionHandler()
```

Comment:
We should return an error on the completion handler, as it's done in the Deferred method. We can use the `Result` type to do this, as it was done in other related PRs.